
 This patch fixes (hopefully) a segment fault in restore when an
 empty path is found for a file.
 Apply it to 1.36.0 with:

 cd <bacula-source>
 patch -p0 <1.36.0-tree.patch
 make
 make install

Index: src/lib/tree.c
===================================================================
RCS file: /cvsroot/bacula/bacula/src/lib/tree.c,v
retrieving revision 1.18
diff -u -r1.18 tree.c
--- src/lib/tree.c	16 Jul 2004 07:23:40 -0000	1.18
+++ src/lib/tree.c	21 Nov 2004 16:03:53 -0000
@@ -87,6 +87,7 @@
    }
    Dmsg2(400, "count=%d size=%d\n", count, size);
    malloc_buf(root, size);
+   root->cached_path_len = -1;
    root->cached_path = get_pool_memory(PM_FNAME);
    root->type = TN_ROOT;
    root->fname = "";
